src/core/opamSystem.ml

```diff
@@ -1015,6 +1009,22 @@ let link src dst =
       copy_dir src dst
     else
       copy_file src dst
+  in
+  mkdir (Filename.dirname dst);
+  if file_or_symlink_exists dst then (
+    if Sys.win32 then Unix.chmod dst 0o640 else ();
```

Comment:
```suggestion
    (* TODO: move this into remove_file *)
    if Sys.win32 then Unix.chmod dst 0o640;
```
the else branch is optional if the return type is unit.
I also added a comment to help see what’s left to be done